spyder/workers/updates.py

```diff
@@ -225,16 +227,38 @@ def _progress_reporter(self, block_number, read_size, total_size):
     def _download_installer(self):
         """Donwload latest Spyder standalone installer executable."""
         tmpdir = tempfile.gettempdir()
-        is_full_installer = (is_module_installed('numpy') or
-                             is_module_installed('pandas'))
-        if os.name == 'nt':
-            name = 'Spyder_64bit_{}.exe'.format('full' if is_full_installer
-                                                else 'lite')
+
+        # If running on macOS under Rosetta, platform.machine will not be
+        # correct so check for ARM64 in version instead. All other
+        # platforms will be x86_64.
+        mach = "arm64" if "ARM64" in platform.version() else "x86_64"
+
+        if parse(self.latest_release_version) > parse("5"):
+            # conda-based installer name
+            if os.name == 'nt':
+                plat, ext = 'Windows', 'exe'
+            elif platform.system == 'Darwin':
+                plat, ext = 'macOS', 'pkg'
+            else:
+                plat, ext = 'Linux', 'sh'
+            name = f'Spyder-{plat}-{mach}.{ext}'
         else:
-            name = 'Spyder{}.dmg'.format('' if is_full_installer else '-Lite')
+            # 5.x installer name
+            is_full_installer = (is_module_installed('numpy') or
+                                 is_module_installed('pandas'))
+            if os.name == 'nt':
+                name = 'Spyder_64bit_{}.exe'.format(
+                    'full' if is_full_installer else 'lite'
+                )
+            else:
+                name = 'Spyder{}{}.dmg'.format(
+                    '_{mach}' if mach == 'arm64' else '',
+                    '' if is_full_installer else '-Lite'
+                )
+
+        url = ('https://github.com/spyder-ide/spyder/releases/download/'
+               f'v{self.latest_release_version}/{name}')
```

Comment:
Just curious on the reason to change this? Concern that the most recent release shown by GitHub wouldn't match the one with the highest version number in the tag as returned by the GitHub releases API?

Reply:
I suppose it does not need to be changed. I changed it on master because non-stable releases are offered for download. For 5.x it is not as critical.

Reply:
Ah, I see. In any case, seems more robust with this change to specify the version to update to explicitly, rather than rely on GitHub's implicit heuristics to always show the one we want. So +1 on the change.